### PR TITLE
docs(label): add documentation and example for text recolor

### PR DESCRIPTION
**Pull request by lhdjply**
docs(label): add documentation and example for text recolor

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by C47D — APPROVED** [review]

File: docs/details/widgets/label.rst [hunk 1]
```diff
@@ -117,6 +117,18 @@ this implementation detail is unnoticed. This is not the case with
...
+
+Text recolor
+------------
+
+In the text, you can use commands to recolor parts of the text. 
+For example: ``Write a #ff0000 red# word``. This feature can be enabled 
+individually for each label by :cpp:expr:`lv_label_set_recolor(label, en)`
+function, recoloring is only supported when the text wrapped with 
+``##ff0000 ... #`` sintax is in one line, it is not supported in wrapped text, 
+see example ``Line wrap, recoloring and scrolling``.
```
Comment (liamHowatt) [inline docs/details/widgets/label.rst]:
```suggestion
function. Recoloring is only supported when the text is wrapped with 
``#ff0000 ... #`` syntax is in one line. It is not supported in wrapped text. 
See example ``Line wrap, recoloring and scrolling``.
```

**Review by liamHowatt — COMMENTED** [review]
Nice!

**Comment by liamHowatt** [comment]
cc @vwheeler63

File: docs/details/widgets/label.rst (hunk 1 above)
Comment (lhdjply) [reply docs/details/widgets/label.rst]: Thank you!

**Review by liamHowatt — DISMISSED** [review]

File: docs/details/widgets/label.rst [hunk 2]
```diff
@@ -117,6 +117,18 @@ this implementation detail is unnoticed. This is not the case with
...
+
+Text recolor
+------------
+
+In the text, you can use commands to recolor parts of the text. 
+For example: ``Write a #ff0000 red# word``. This feature can be enabled 
+individually for each label by :cpp:expr:`lv_label_set_recolor(label, en)`
+function. Recoloring is only supported when the text is wrapped with 
+``#ff0000 ... #`` syntax is in one line. It is not supported in wrapped text. 
+See example ``Line wrap, recoloring and scrolling``.
```
Comment (vwheeler63) [inline docs/details/widgets/label.rst]:
```suggestion
function. In the context of word-wrapped text, any Recoloring started on a
line will be terminated at the end of the line where the line is wrapped if it
was not already terminated by an ending ``#`` in the text.
```

**Review by vwheeler63 — CHANGES_REQUESTED** [review]
Nice!

**Review by liamHowatt — APPROVED** [review]

File: docs/details/widgets/label.rst (hunk 2 above)
Comment (liamHowatt) [reply docs/details/widgets/label.rst]: Thanks Vic!

**Review by kisvegabor — APPROVED** [review]
Thank you!

**Pull request merged**